### PR TITLE
Adaptations for Qt 5.13

### PR DESCRIPTION
**Pull request by chrisderock**
This patch replaces "qSort" with "std::sort" and "QFontMetrics::width" with "QFontMetrics::horizontalAdvance" if compiled with QT 5.13.

### CODE REVIEW

**Comment by mcallegari** [comment]
Thanks, but I don't like how you did the sort change.
I'd rather put a define in a common .h file (e.g. engine/src/utils.h), like:
```
#if (QT_VERSION >= QT_VERSION_CHECK(5,13,0))
#define qSort(b, e, c)  std::sort(b, e, c)
#endif
```

File: engine/src/sequence.cpp [hunk 1]
```diff
@@ -169,7 +169,11 @@ bool Sequence::loadXML(QXmlStreamReader &root)
     if (scene != NULL)
     {
         sceneValues = scene->values();
+#if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
         qSort(sceneValues.begin(), sceneValues.end());
+#else
+	(sceneValues.begin(), sceneValues.end());
```
Comment (mcallegari) [inline engine/src/sequence.cpp]: this won't even build

**Comment by chrisderock** [comment]
i think that would be useful for this special situation. but how to definde this for qSort(container) vs. std::sort(container.begin(), container.end())?

File: engine/src/sequence.cpp (hunk 1 above)
Comment (chrisderock) [reply engine/src/sequence.cpp]: sorry, my fault. i forgot the function call...

**Comment by mcallegari** [comment]
> i think that would be useful for this special situation. but how to definde this for qSort(container) vs. std::sort(container.begin(), container.end())?

You just change `container` to `container.begin(), container.end()`

**Comment by chrisderock** [comment]
i can change the modifications as you sugested.. but in what for i file should i use? this covers the whole source tree?

**Comment by mcallegari** [comment]
...or we can just use std:sort eveywhere.
I think qSort was just a wrapper anyway, and std:sort could have been used since the beginning, even during Qt4 times.
My only concern is toolchain compatibility, but apparently std:sort if C++98, so should be fine

**Comment by chrisderock** [comment]
i removed the version check but in rgbtext.cpp and rgbtext_test.cpp it is needed (imho)

File: engine/src/rgbtext.cpp [hunk 2]
```diff
@@ -161,8 +161,13 @@ int RGBText::scrollingTextStepCount() const
     QFontMetrics fm(m_font);
     if (animationStyle() == Vertical)
         return m_text.length() * fm.ascent();
-    else
+    else{
+#if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
         return fm.width(m_text);
+#else
+	return fm.horizontalAdvance(m_text);
```
Comment (mcallegari) [inline engine/src/rgbtext.cpp]: Indentation please

File: qmlui/importmanager.cpp [hunk 3]
```diff
@@ -439,7 +439,7 @@ void ImportManager::importFunctionID(quint32 funcID)
 
             if (removeList.isEmpty() == false)
             {
-                qSort(removeList.begin(), removeList.end());
+		std::sort(removeList.begin(), removeList.end());
```
Comment (mcallegari) [inline qmlui/importmanager.cpp]: Indentation

File: qmlui/virtualconsole/vcslider.cpp [hunk 4]
```diff
@@ -606,8 +606,8 @@ void VCSlider::slotTreeDataChanged(TreeModelItem *item, int role, const QVariant
     else
     {
         addLevelChannel(fixtureID, chIndex);
-        qSort(m_levelChannels.begin(), m_levelChannels.end());
-    }
+	std::sort(m_levelChannels.begin(), m_levelChannels.end());
```
Comment (mcallegari) [inline qmlui/virtualconsole/vcslider.cpp]: Indentation

File: qmlui/virtualconsole/vcslider.cpp [hunk 5]
```diff
@@ -1381,9 +1381,9 @@ bool VCSlider::loadXMLLevel(QXmlStreamReader &level_root)
         } while (level_root.readNextStartElement());
     }
 
-    if (m_levelChannels.count())
-        qSort(m_levelChannels.begin(), m_levelChannels.end());
-
+    if (m_levelChannels.count()){
+	std::sort(m_levelChannels.begin(), m_levelChannels.end());
```
Comment (mcallegari) [inline qmlui/virtualconsole/vcslider.cpp]: Indentation. No parenthesis

File: ui/src/rgbmatrixeditor.cpp [hunk 6]
```diff
@@ -1067,7 +1067,7 @@ void RGBMatrixEditor::slotSaveToSequenceClicked()
             }
             // !! Important !! matrix's heads can be displaced randomly but in a sequence
             // we absolutely need ordered values. So do it now !
-            qSort(step.values.begin(), step.values.end());
+	    std::sort(step.values.begin(), step.values.end());
```
Comment (mcallegari) [inline ui/src/rgbmatrixeditor.cpp]: Indentation

File: ui/src/virtualconsole/vcslider.cpp [hunk 7]
```diff
@@ -549,7 +549,7 @@ void VCSlider::addLevelChannel(quint32 fixture, quint32 channel)
     if (m_levelChannels.contains(lch) == false)
     {
         m_levelChannels.append(lch);
-        qSort(m_levelChannels.begin(), m_levelChannels.end());
+	std::sort(m_levelChannels.begin(), m_levelChannels.end());
```
Comment (mcallegari) [inline ui/src/virtualconsole/vcslider.cpp]: Indentation

File: engine/src/rgbtext.cpp (hunk 2 above)
Comment (chrisderock) [reply engine/src/rgbtext.cpp]: Can you add a comment about the indentation style maybe to the README.md?

**Comment by mcallegari** [comment]
Looks ok now, thanks

**Pull request merged**